### PR TITLE
Bump imagemin version

### PR DESCRIPTION
**Pull request by kevva**
This will also remove the functionality for automatically detecting directories since all it did was bloat the code imo. Also renamed the shorthand for `optimizationLevel` to `O` instead of `o` which is replaced by `outdir`. `O` is also pretty common among the binaries we're using.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+});
+
+test('show version', async t => {
+	t.is(await execa.stdout('./cli.js', ['--version']), require('./package.json').version);
+});
+
+test('optimize a GIF', async t => {
+	const buf = await fsP.readFile(path.join(__dirname, 'fixtures', 'test.gif'));
+	const cp = execa('./cli.js');
+	cp.stdin.end(buf);
```
Comment (sindresorhus) [inline test.js]: Use the `input` option ;)

Comment (sindresorhus) [inline test.js]: also `execa.stdout()`

Comment (kevva) [inline test.js]: Woot, yeah, forgot about the option T_T.

File: cli.js [hunk 2]
```diff
@@ -1,30 +1,27 @@
...
-const pathExists = require('path-exists');
-const Imagemin = require('imagemin');
+const imagemin = require('imagemin');
+const stripIndent = require('strip-indent');
 
 const cli = meow(`
 	Usage
-	  $ imagemin <file> <directory>
-	  $ imagemin <directory> <output>
+	  $ imagemin <path|glob> --outdir build
```
Comment (sindresorhus) [inline cli.js]:
```
$ imagemin <path|glob> --outdir=build
```

and shouldn't it be `--out-dir`?

File: cli.js [hunk 3]
```diff
@@ -1,30 +1,27 @@
...
 	  $ imagemin <file> > <output>
 	  $ cat <file> | imagemin > <output>
 	  $ imagemin [--plugin <plugin-name>...] ...
 
 	Options
+	  -O, --optimizationLevel <number>  Optimization level between 0 and 7
 	  -P, --plugin                      Override the default plugins
```
Comment (sindresorhus) [inline cli.js]: Have you thought about supporting plugin options?

File: cli.js (hunk 2 above)
Comment (sindresorhus) [inline cli.js]: And why did you change it to a flag?

File: cli.js [hunk 4]
```diff
@@ -36,117 +33,64 @@ const cli = meow(`
...
 	'optipng',
 	'svgo'
 ];
 
-function run(src, dest) {
-	const plugins = cli.flags.plugin ? arrify(cli.flags.plugin) : DEFAULT_PLUGINS;
-	const imagemin = new Imagemin().src(src);
+const requirePlugins = (plugins, opts) => plugins.map(x => {
+	try {
+		x = require(`imagemin-${x}`)(opts);
```
Comment (sindresorhus) [inline cli.js]: You can just return the require directly here.

File: cli.js [hunk 5]
```diff
@@ -36,117 +33,64 @@ const cli = meow(`
...
+	} catch (err) {
+		console.error(stripIndent(`
+			Unknown plugin: ${x}
 
-	plugins.forEach(name => {
-		let plugin;
+			Maybe you forgot to install the plugin?
```
Comment (sindresorhus) [inline cli.js]: Maybe => Did

File: cli.js [hunk 6]
```diff
@@ -36,117 +33,64 @@ const cli = meow(`
...
-		imagemin.use(plugin);
-	});
+	return x;
+});
 
-	if (dest) {
-		imagemin.dest(dest);
-	}
+const run = (input, opts) => {
+	opts.plugin = opts.plugin ? arrify(opts.plugin) : DEFAULT_PLUGINS;
```
Comment (sindresorhus) [inline cli.js]: Don't modify the user-supplied object. This will cause havok if the users uses the same object for multiple calls.

File: cli.js (hunk 2 above)
Comment (kevva) [inline cli.js]: > and shouldn't it be --out-dir?

I'm ok with whatever.

> And why did you change it to a flag?

So that you can specify multiple input files.

Comment (sindresorhus) [inline cli.js]: Ok, but the docs doesn't indicate that's possible.

File: cli.js (hunk 4 above)
Comment (kevva) [inline cli.js]: Yeah I know, but `xo` complains since it's inside a "block" and therefore nothing is returned according to `xo`.

Comment (sindresorhus) [inline cli.js]: Sounds like the rule could be improved. There's no way for it to not return because of the `process.exit()`. It should detect that and ignore. Mind opening an ESLint ticket?

This should be possible without a violation:

```js
arr.map(() => {
	try {
		return require('foo');
	} catch (err) {
		console.error(err);
		process.exit(1);
	}
});
```

Comment (kevva) [inline cli.js]: I think I've encountered this in some other case too. Don't remember it though.

File: cli.js (hunk 3 above)
Comment (kevva) [inline cli.js]: Like `imagemin -P [pngquant --option]`? Sure, but I rather do it in another PR.

File: cli.js [hunk 7]
```diff
@@ -1,30 +1,27 @@
...
 	  $ cat <file> | imagemin > <output>
 	  $ imagemin [--plugin <plugin-name>...] ...
 
 	Options
+	  -O, --optimizationLevel <number>  Optimization level between 0 and 7
 	  -P, --plugin                      Override the default plugins
 	  -i, --interlaced                  Interlace gif for progressive rendering
-	  -o, --optimizationLevel <number>  Optimization level between 0 and 7
+	  -o, --out-dir                     Output directory
 	  -p, --progressive                 Lossless conversion to progressive
```
Comment (sindresorhus) [inline cli.js]: I think we should do what we did in gulp-imagemin and drop all the plugin options here. We can rather introduce ability to supply plugin options directly to the plugins in another PR.

File: cli.js (hunk 3 above)
Comment (sindresorhus) [inline cli.js]: I dunno. Can you open a new issue so we can find a good solution for it?

File: cli.js (hunk 7 above)
Comment (kevva) [inline cli.js]: Yup, good idea.

**Comment by sindresorhus** [comment]
Looks good. Nice work @kevva :)
